File: go/fiber/internal/modules/admin/controller.go
package admin

import (
	"strconv"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/admin/dto"

	"github.com/gofiber/fiber/v2"
)

type AdminController struct{
	service AdminService
}

func (c *AdminController) FindAllManagers(ctx *fiber.Ctx) error {
	quantidade := ctx.Query("quantidade")

	quantidadeInt, _ := strconv.Atoi(quantidade)

	dto := dto.FindAllQueryDto{
		Quantidade: quantidadeInt,
	}

	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	managers, err := c.service.FindAllManagers(dto.Quantidade)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusOK).JSON(managers)
}

func (c *AdminController) FindAllClients(ctx *fiber.Ctx) error {
	quantidade := ctx.Query("quantidade")

	quantidadeInt, _ := strconv.Atoi(quantidade)

	dto := dto.FindAllQueryDto{
		Quantidade: quantidadeInt,
	}

	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	clients, err := c.service.FindAllClients(dto.Quantidade)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusOK).JSON(clients)
}

File: rust/axum/src/modules/admin/controller.rs
use axum::{
    extract::Query,
    response::IntoResponse,
    Json
};
use validator::Validate;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;

use super::dto::FindAllQueryDto;
use super::service::AdminService;

pub struct AdminController{
}

impl AdminController {
    pub async fn find_all_managers(
        Query(dto): Query<FindAllQueryDto>
    ) -> impl IntoResponse {
        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        match AdminService::find_all_managers(dto.quantidade).await {
            Ok(managers) => Json(managers).into_response(),
            Err(err) => err.into_response(),
        }
    }

    pub async fn find_all_clients(
        Query(dto): Query<FindAllQueryDto>
    ) -> impl IntoResponse {
        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        match AdminService::find_all_clients(dto.quantidade).await {
            Ok(clients) => Json(clients).into_response(),
            Err(err) => err.into_response(),
        }
    }
}
